### PR TITLE
fix: support for nodenext module

### LINKED ISSUE #1499: [bug] Module '"@rainbow-me/rainbowkit"' has no exported member 'RainbowKitProvider'.ts(2305)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### RainbowKit Version

1.0.11

### wagmi Version

1.3.11

### Current Behavior

Trying to import `RainbowKitProvider` from '@rainbow-me/rainbowkit'

### Expected Behavior

The import works

### Steps To Reproduce

Try importing in a package with
```
"module": "NodeNext",
"moduleResolution": "NodeNext",
```

### Link to Minimal Reproducible Example (CodeSandbox, StackBlitz, etc.)

https://github.com/yearn/web-lib/pull/322

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by 0xmkh**
In RainbowKit, the `index.ts` file re-exports components. But when you use `NodeNext` in TypeScript, there can be problems because `NodeNext` has its own rules and limits for how modules should work.

### CODE REVIEW

**Comment by DanielSinclair** [comment]
Seeing some build errors when using `NodeNext` in all of the tsconfigs: 
```
packages/rainbowkit prepare$ MINIFY_CSS=true pnpm build
[701 lines collapsed]
│ src/wallets/walletConnectors/zealWallet/zealWallet.ts(4,24): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' …
│ src/wallets/walletConnectors/zealWallet/zealWallet.ts(5,37): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' …
│ src/wallets/walletConnectors/zerionWallet/zerionWallet.ts(3,23): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResoluti…
│ src/wallets/walletConnectors/zerionWallet/zerionWallet.ts(4,37): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResoluti…
│ src/wallets/walletConnectors/zerionWallet/zerionWallet.ts(5,23): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResoluti…
│ src/wallets/walletConnectors/zerionWallet/zerionWallet.ts(6,24): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResoluti…
│ src/wallets/walletConnectors/zerionWallet/zerionWallet.ts(7,43): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResoluti…
│ src/wallets/walletConnectors/zerionWallet/zerionWallet.ts(11,8): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResoluti…
│ > @rainbow-me/rainbowkit@1.3.2 build /Users/daniel/Desktop/rainbowkit/packages/rainbowkit
│ > node build.js
└─ Done in 2.8s
. postinstall$ husky install
│ husky - Git hooks installed
└─ Done in 362ms
packages/rainbowkit-siwe-next-auth prepare$ pnpm build
[4 lines collapsed]
│ src/index.ts(1,48): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean '…
│ src/index.ts(2,44): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean '…
│ src/RainbowKitSiweNextAuthProvider.tsx(38,27): error TS7031: Binding element 'address' implicitly has an 'any' type.
│ src/RainbowKitSiweNextAuthProvider.tsx(38,36): error TS7031: Binding element 'chainId' implicitly has an 'any' type.
│ src/RainbowKitSiweNextAuthProvider.tsx(38,45): error TS7031: Binding element 'nonce' implicitly has an 'any' type.
│ src/RainbowKitSiweNextAuthProvider.tsx(63,28): error TS7031: Binding element 'message' implicitly has an 'any' type.
│ src/RainbowKitSiweNextAuthProvider.tsx(75,26): error TS7031: Binding element 'message' implicitly has an 'any' type.
│ src/RainbowKitSiweNextAuthProvider.tsx(75,35): error TS7031: Binding element 'signature' implicitly has an 'any' type.
│ > @rainbow-me/rainbowkit-siwe-next-auth@0.3.4 build /Users/daniel/Desktop/rainbowkit/packages/rainbowkit-siwe-next-auth
│ > node build.js
└─ Done in 1.3s
packages/rainbow-button prepare$ pnpm build
│ > @rainbow-me/rainbow-button@0.1.2 prebuild /Users/daniel/Desktop/rainbowkit/packages/rainbow-button
│ > pnpm typegen
│ > @rainbow-me/rainbow-button@0.1.2 typegen /Users/daniel/Desktop/rainbowkit/packages/rainbow-button
│ > tsc --emitDeclarationOnly || true
│ src/components/RainbowButton.tsx(2,46): error TS2307: Cannot find module '@rainbow-me/rainbowkit/dist/components/RainbowKitProvider/RainbowKitProvider' or its corresponding ty…
│ src/index.ts(1,39): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean '…
│ src/index.ts(2,34): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean '…
│ src/index.ts(3,31): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean '…
│ > @rainbow-me/rainbow-button@0.1.2 build /Users/daniel/Desktop/rainbowkit/packages/rainbow-button
│ > node build.js
└─ Done in 1.3s
```

File: packages/create-rainbowkit/generated-test-app/pages/_app.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import React from 'react';
```
Comment (DanielSinclair) [inline packages/create-rainbowkit/generated-test-app/pages/_app.tsx:1]: Was there a reason we needed this change?

Comment (KosmosKey) [reply packages/create-rainbowkit/generated-test-app/pages/_app.tsx:1]: Yeah unless i change something in tsconfig.json:

image

**Comment by KosmosKey** [comment]
> Seeing some build errors when using `NodeNext` in all of the tsconfigs:
> 
> ```
> packages/rainbowkit prepare$ MINIFY_CSS=true pnpm build
> [701 lines collapsed]
> │ src/wallets/walletConnectors/zealWallet/zealWallet.ts(4,24): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' …
> │ src/wallets/walletConnectors/zealWallet/zealWallet.ts(5,37): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' …
> │ src/wallets/walletConnectors/zerionWallet/zerionWallet.ts(3,23): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResoluti…
> │ src/wallets/walletConnectors/zerionWallet/zerionWallet.ts(4,37): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResoluti…
> │ src/wallets/walletConnectors/zerionWallet/zerionWallet.ts(5,23): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResoluti…
> │ src/wallets/walletConnectors/zerionWallet/zerionWallet.ts(6,24): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResoluti…
> │ src/wallets/walletConnectors/zerionWallet/zerionWallet.ts(7,43): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResoluti…
> │ src/wallets/walletConnectors/zerionWallet/zerionWallet.ts(11,8): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResoluti…
> │ > @rainbow-me/rainbowkit@1.3.2 build /Users/daniel/Desktop/rainbowkit/packages/rainbowkit
> │ > node build.js
> └─ Done in 2.8s
> . postinstall$ husky install
> │ husky - Git hooks installed
> └─ Done in 362ms
> packages/rainbowkit-siwe-next-auth prepare$ pnpm build
> [4 lines collapsed]
> │ src/index.ts(1,48): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean '…
> │ src/index.ts(2,44): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean '…
> │ src/RainbowKitSiweNextAuthProvider.tsx(38,27): error TS7031: Binding element 'address' implicitly has an 'any' type.
> │ src/RainbowKitSiweNextAuthProvider.tsx(38,36): error TS7031: Binding element 'chainId' implicitly has an 'any' type.
> │ src/RainbowKitSiweNextAuthProvider.tsx(38,45): error TS7031: Binding element 'nonce' implicitly has an 'any' type.
> │ src/RainbowKitSiweNextAuthProvider.tsx(63,28): error TS7031: Binding element 'message' implicitly has an 'any' type.
> │ src/RainbowKitSiweNextAuthProvider.tsx(75,26): error TS7031: Binding element 'message' implicitly has an 'any' type.
> │ src/RainbowKitSiweNextAuthProvider.tsx(75,35): error TS7031: Binding element 'signature' implicitly has an 'any' type.
> │ > @rainbow-me/rainbowkit-siwe-next-auth@0.3.4 build /Users/daniel/Desktop/rainbowkit/packages/rainbowkit-siwe-next-auth
> │ > node build.js
> └─ Done in 1.3s
> packages/rainbow-button prepare$ pnpm build
> │ > @rainbow-me/rainbow-button@0.1.2 prebuild /Users/daniel/Desktop/rainbowkit/packages/rainbow-button
> │ > pnpm typegen
> │ > @rainbow-me/rainbow-button@0.1.2 typegen /Users/daniel/Desktop/rainbowkit/packages/rainbow-button
> │ > tsc --emitDeclarationOnly || true
> │ src/components/RainbowButton.tsx(2,46): error TS2307: Cannot find module '@rainbow-me/rainbowkit/dist/components/RainbowKitProvider/RainbowKitProvider' or its corresponding ty…
> │ src/index.ts(1,39): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean '…
> │ src/index.ts(2,34): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean '…
> │ src/index.ts(3,31): error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean '…
> │ > @rainbow-me/rainbow-button@0.1.2 build /Users/daniel/Desktop/rainbowkit/packages/rainbow-button
> │ > node build.js
> └─ Done in 1.3s
> ```

Huh weird. Can you try now ? I deleted the `index.ts` from `/components`. All builds works for me:

image

**Review by DanielSinclair — APPROVED** [review]

**Pull request merged**